sdk/iot/provisioning/tests/cmocka/test_az_iot_provisioning_client.c

```diff
@@ -75,60 +91,162 @@ static void test_az_iot_provisioning_client_custom_options_get_username_succeed(
       &options);
   assert_int_equal(AZ_OK, ret);
 
-  char user_name[128];
-  memset(user_name, 0xCC, sizeof(user_name));
-
-  char* expected_username = TEST_ID_SCOPE
+  char expected_username[] = TEST_ID_SCOPE
       "/registrations/" TEST_REGISTRATION_ID
       "/api-version=" AZ_IOT_PROVISIONING_SERVICE_VERSION USER_AGENT_PREFIX TEST_USER_AGENT;
 
-  ret = az_iot_provisioning_client_get_user_name(&client, user_name, sizeof(user_name), NULL);
+  char user_name[sizeof(expected_username) + 1];
+  memset(user_name, 0xCC, sizeof(user_name));
+  size_t user_name_len;
+  ret = az_iot_provisioning_client_get_user_name(
+      &client, user_name, sizeof(user_name), &user_name_len);
   assert_int_equal(AZ_OK, ret);
   assert_memory_equal(expected_username, user_name, strlen(user_name));
   assert_int_equal((uint8_t)0xCC, (uint8_t)user_name[strlen(user_name) + 1]);
+  assert_int_equal(strlen(expected_username), user_name_len);
 }
 
-static void test_az_iot_provisioning_client_get_subscribe_topic_filter_succeed()
+static void test_az_iot_provisioning_client_get_connect_info_insufficient_space_fails()
 {
   az_iot_provisioning_client client;
-  char topic[128];
-  memset(topic, 0xCC, sizeof(topic));
+  az_iot_provisioning_client_options options = az_iot_provisioning_client_options_default();
+  options.user_agent = AZ_SPAN_FROM_STR(TEST_USER_AGENT);
+
+  az_result ret = az_iot_provisioning_client_init(
+      &client,
+      test_global_device_endpoint,
+      AZ_SPAN_FROM_STR(TEST_ID_SCOPE),
+      AZ_SPAN_FROM_STR(TEST_REGISTRATION_ID),
+      &options);
+  assert_int_equal(AZ_OK, ret);
 
-  char* expected_topic = "$dps/registrations/res/#";
+  char client_id[sizeof(TEST_REGISTRATION_ID) - 1];
+  memset(client_id, 0xCC, sizeof(client_id));
+  size_t client_id_len = 0xBAADC0DE;
```

Comment:
Nice :p

3131949278 > int32.max (0x7FFFFFFF)

Reply:
The idea here is to verify it's not changed if we error out.